stream_alert/rule_processor/handler.py

```diff
@@ -133,6 +160,69 @@ def get_alerts(self):
         """
         return self._alerts
 
+    def _send_to_firehose(self):
+        """Put all classified records into its respective Firehose Delivery Stream
+
+        Args:
+            categorized_payloads (defaultdict): Categorized set of payloads by classified type
+        """
+        def _chunk(record_list, chunk_size):
+            """Helper function to chunk payloads"""
+            for item in range(0, len(record_list), chunk_size):
+                yield record_list[item:item + chunk_size]
+
+        def _check_record_batch(batch):
+            """Helper function to verify record size"""
+            for index, record in enumerate(batch):
+                if len(str(record).encode('utf-8')) > MAX_RECORD_SIZE:
+                    # Show the first 1k bytes in order to not overload
+                    # CloudWatch logs
+                    LOGGER.error('The following record is too large'
+                                 'be sent to Firehose: %s', str(record)[:1000])
+                    MetricLogger.log_metric(FUNCTION_NAME,
+                                            MetricLogger.FIREHOSE_FAILED_RECORDS,
+                                            1)
+                    batch.pop(index)
+
+        delivery_stream_name_pattern = 'streamalert_data_{}'
+
+        # Iterate through each payload type
+        for log_type, records in self.categorized_payloads.items():
+            # This same method is used when naming the Delivery Streams
+            formatted_log_type = log_type.replace(':', '_')
+
+            for record_batch in _chunk(records, MAX_BATCH_SIZE):
+                stream_name = delivery_stream_name_pattern.format(formatted_log_type)
+                _check_record_batch(record_batch)
+
+                resp = self.firehose_client.put_record_batch(
+                    DeliveryStreamName=stream_name,
+                    Records=[{'Data': json.dumps(record, separators=(",", ":")) + '\n'}
```

Comment:
Can you add a comment about the newline that's being appended here? Curious why this would be needed? Since strings are immutable, this will be creating a copy of the string dumped json every time just to add a new line to it, causing potentially unwanted memory usage and processing time. Just something to keep in mind if is nice to have but not required

Reply:
It's required by Firehose or else all records will be written to a single line

Reply:
Can you add a comment saying it's required? I know I'll see that in the future and totally nix it without a comment